testutil/db/db.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+
+	"github.com/prometheus/tsdb"
+	"github.com/prometheus/tsdb/labels"
+	"github.com/prometheus/tsdb/testutil"
+	"github.com/prometheus/tsdb/tsdbutil"
+)
+
+// CreateBlock creates a block with given set of series and returns its dir.
+func CreateBlock(tb testing.TB, dir string, series []tsdb.Series) string {
```

Comment:
I agree! This is going to be a bigger change though since just using the new function in the old test leads to import cycles (`tsdb` -> `testutil/db` -> `tsdb`). I'll work on testing the `analyze` function first, might be more functions that need to be moved out. I'll try to think of a solution after that, most likely involving some interface. Let me know if you have any suggestions!

Reply:
in the past I have used [this](https://stackoverflow.com/questions/39028261/import-cycle-in-golang-with-test-packages). Not sure if it will help here.

Reply:
I looked at that possibility but moving the existing tests into a new package would require moving lots of stuff around which seemed very nasty. The easiest way seems to be to just move `CreateBlock` and `GenSeries` back into the `tsdb` package (new file: `mocks.go` ) but mark them clearly as test functions: `MockCreateBlock` and `MockGenSeries`. I have also moved dependent functions and types into `mocks.go` and deleted them where they were defined previously.